addons/hr_expense/models/hr_expense.py

```diff
@@ -312,3 +286,130 @@ def action_get_attachment_view(self):
+                total += line.total_amount
+            sheet.total_amount = total
+
+    @api.multi
+    def _compute_attachment_number(self):
+        for sheet in self:
+            total = 0.0
+            for line in sheet.expense_line_ids:
+                total += line.attachment_number
+            sheet.attachment_number = total
+
+    @api.multi
+    def refuse_expenses(self, reason):
+        self.write({'state': 'cancel'})
+        if self.employee_id.user_id:
+            body = (_("Your Expense %s has been refused.<br/><ul class=o_timeline_tracking_value_list><li>Reason<span> : </span><span class=o_timeline_tracking_value>%s</span></li></ul>") % (self.name, reason))
+            self.message_post(body=body, partner_ids=[self.employee_id.user_id.partner_id.id])
+
+    @api.multi
+    def paid_expenses(self):
+        self.write({'state': 'done'})
+
+    @api.multi
+    def reset_expenses(self):
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def submit_expense_sheets(self):
+        if any(sheet.state != 'draft' for sheet in self):
+            raise UserError(_("You can only submit draft sheets!"))
+        self.write({'state': 'submit'})
+
+    @api.multi
+    def approve_expense_sheets(self):
+        self.write({'state': 'approve', 'user_valid': self._uid})
+
+    @api.multi
+    def refuse_expense_sheets(self, reason):
+        self.write({'state': 'cancel'})
+        if self.employee_id.user_id:
+            body = (_("Your Expense %s has been refused.<br/><ul class=o_timeline_tracking_value_list><li>Reason<span> : </span><span class=o_timeline_tracking_value>%s</span></li></ul>") % (self.name, reason))
+            self.message_post(body=body, partner_ids=[self.employee_id.user_id.partner_id.id])
+
+    @api.multi
+    def paid_expense_sheets(self):
+        self.write({'state': 'done'})
+
+    @api.multi
+    def reset_expense_sheets(self):
+        return self.write({'state': 'draft'})
+
+    @api.multi
+    def action_sheet_move_create(self):
+        if any(sheet.state != 'approve' for sheet in self):
+            raise UserError(_("You can only generate accounting entry for approved expense(s)."))
+
+        if any(sheet.employee_id != self[0].employee_id for sheet in self):
+            raise UserError(_("Expenses must belong to the same Employee."))
+
+        if any(not expense.journal_id for expense in self.mapped('expense_line_ids')):
+            raise UserError(_("Expenses must have an expense journal specified to generate accounting entries."))
+
+        res = self.mapped('expense_line_ids').action_move_create()
+        self.write({'state': 'post'})
+        return res
+
+    @api.multi
+    def action_get_attachment_view(self):
+        self.ensure_one()
+        res = self.env['ir.actions.act_window'].for_xml_id('base', 'action_attachment')
+        res['domain'] = [('res_model', '=', 'hr.expense'), ('res_id', 'in', self.mapped('expense_line_ids.id'))]
```

Comment:
Actually you could imply the ensure_one (and fix your mapped by the way) by using self.expense_line_ids.ids .